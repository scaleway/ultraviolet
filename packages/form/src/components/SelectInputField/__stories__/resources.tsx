export const planets = {
  'terrestrial planets': [
    {
      value: 'mercury',
      label: 'Mercury',
      disabled: false,
    },
    {
      value: 'venus',
      label: 'Venus',
      disabled: false,
    },
    {
      value: 'earth',
      label: 'Earth',
      disabled: false,
      description: 'Our home planet',
      searchText: 'earth',
    },
    {
      value: 'mars',
      label: 'Mars',
      disabled: true,
    },
    {
      value: 'pluto',
      label: 'Pluto',
      disabled: false,
      description:
        'Pluto does not fit the usual classification of either terrestrial or Jovian planets, but is rocky',
    },
  ],

  'jovian planets': [
    {
      value: 'jupiter',
      label: 'Jupiter',
      disabled: false,
      description:
        'Jupiter is the fifth planet from the Sun and the largest in the Solar System. It is a gas giant with a mass more than two and a half times that of all the other planets in the Solar System combined, and slightly less than one one-thousandth the mass of the Sun.',
    },
    {
      value: 'saturn',
      label: 'Saturn',
      disabled: false,
    },
    {
      value: 'uranus',
      label: 'Uranus',
      disabled: false,
    },

    {
      value: 'neptune',
      label: 'Neptune',
      disabled: false,
    },
  ],
}

export const cities = [
  {
    value: 'par',
    label: 'Paris',
    disabled: false,
    optionalInfo: '12',
    description: 'France',
    searchText: 'paris',
  },
  {
    value: 'ams',
    label: 'Amsterdam',
    disabled: false,
    optionalInfo: '14',
    searcText: 'amsterdam',
  },
  {
    value: 'waw',
    label: 'Warsaw',
    disabled: false,
    optionalInfo: '1',
    searchText: 'warsaw',
  },
]
