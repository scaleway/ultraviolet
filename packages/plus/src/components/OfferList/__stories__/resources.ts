import { ComponentProps } from 'react'
import { OfferList } from '../OfferList'

export const data = [
  {
    id: 'mercury',
    name: 'Mercury',
    perihelion: 0.307,
    aphelion: 0.467,
  },
  {
    id: 'venus',
    name: 'Venus',
    perihelion: 0.718,
    aphelion: 0.728,
  },
  {
    id: 'home-sweet-home',
    name: 'Earth',
    perihelion: 0.983,
    aphelion: 1.017,
  },
  {
    id: 'mars',
    name: 'Mars',
    perihelion: 1.381,
    aphelion: 1.666,
  },
  {
    id: 'jupiter',
    name: 'Jupiter',
    perihelion: 4.951,
    aphelion: 5.457,
  },
  {
    id: 'saturn',
    name: 'Saturn',
    perihelion: 9.041,
    aphelion: 10.124,
  },
  {
    id: 'uranus',
    name: 'Uranus',
    perihelion: 18.286,
    aphelion: 20.097,
  },
  {
    id: 'id-neptune',
    name: 'Neptune',
    perihelion: 29.81,
    aphelion: 30.33,
  },
]

export const columns: NonNullable<ComponentProps<typeof OfferList>['columns']> =
  [
    { label: 'Solar system Planet' },
    { label: 'Perihelion', width: '200px' },
    { label: 'Aphelion', width: '200px' },
  ]
