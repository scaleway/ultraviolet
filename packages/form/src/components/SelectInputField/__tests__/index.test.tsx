import { act, fireEvent, screen, waitFor } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { renderWithForm } from '@utils/test'
import { describe, expect, test, vi } from 'vitest'
import { SelectInputField } from '..'
import { cities, planets } from './resources'

describe('selectInputField', () => {
  test('should render correctly', () => {
    const { asFragment } = renderWithForm(
      <SelectInputField name="test" options={cities} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly disabled', () => {
    const { asFragment } = renderWithForm(
      <SelectInputField disabled name="test" options={cities} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly multiselect', () => {
    const { asFragment } = renderWithForm(
      <SelectInputField multiselect name="test" options={cities} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should render correctly grouped', () => {
    const { asFragment } = renderWithForm(
      <SelectInputField multiselect name="test" options={planets} />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should display right value on grouped options', async () => {
    const { asFragment } = renderWithForm(
      <SelectInputField name="test" options={planets} searchable={false} />,
    )
    const select = screen.getByTestId('select-input-test')
    await userEvent.click(select)
    fireEvent.keyDown(select, { key: 'ArrowDown', keyCode: 40 })
    const mercury = screen.getByTestId(`option-stack-mercury`)

    await userEvent.click(mercury)
    await userEvent.click(select)

    await waitFor(() => {
      expect(screen.getByTestId(`option-stack-mercury`)).toBeVisible()
    })
    expect(asFragment()).toMatchSnapshot()
  })

  test('should trigger events', async () => {
    const onChange = vi.fn()

    const { asFragment } = renderWithForm(
      <SelectInputField
        name="test"
        onChange={onChange}
        options={planets}
        searchable={false}
      />,
    )
    const select = screen.getByTestId('select-input-test')
    await userEvent.click(select)
    const option = screen.getByTestId('option-stack-mercury')

    await userEvent.click(option)
    await waitFor(() => {
      expect(onChange).toBeCalledTimes(1)
    })
    act(() => select.blur())

    expect(asFragment()).toMatchSnapshot()
  })
})
