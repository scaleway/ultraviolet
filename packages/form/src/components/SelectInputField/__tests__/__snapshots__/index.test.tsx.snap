// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`selectInputField > should display right value on grouped options 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_l_"
          aria-describedby="_r_l_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_l_"
              aria-expanded="true"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8 uv_1ba7kndc uv_1ba7kndf"
              data-disabled="false"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_k_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7kndl uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow20 uv_m4c9ow3c"
              >
                Mercury
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_34__1sbwqkz1i"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
          <div
            class="uv_5kpm8k0 uv_w40vpo4 null"
            id="_r_l_"
            role="dialog"
            style="--uv_ozfz3c0: -5px; --uv_ozfz3c1: 0px; --uv_ozfz3c2:  rotate(180deg); --uv_ozfz3c4: translate3d(0px, 4px, 0); --uv_ozfz3c5: 0ms; --uv_ozfz3c6: translate3d(0px, 4px, 0); --uv_ozfz3c7: 100%; opacity: 1;"
          >
            <div
              class="uv_w40vpo2"
            >
              <div
                class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
              >
                <div
                  class="uv_5kpm8k1 uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u5d"
                  data-grouped="true"
                  id="select-dropdown"
                  role="listbox"
                >
                  <div
                    class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
                  >
                    <div
                      class="uv_5kpm8k5"
                    >
                      <button
                        class=" uv_5kpm8k3"
                        data-selectgroup="false"
                        data-testid="group-0"
                        role="group"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owi uv_m4c9ows uv_m4c9ow1b uv_m4c9ow20 uv_m4c9ow3c"
                          style="--uv_qabug40: left;"
                        >
                          JOVIAN PLANETS
                        </span>
                      </button>
                    </div>
                    <div
                      class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
                      id="items"
                    >
                      <div
                        aria-disabled="false"
                        aria-label="jupiter"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-jupiter"
                        id="option-0"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-jupiter"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Jupiter
                            </span>
                          </div>
                          <span
                            class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9own uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
                            style="--uv_qabug40: left;"
                          >
                            Jupiter is the fifth planet from the Sun and the largest in the Solar System. It is a gas giant with a mass more than two and a half times that of all the other planets in the Solar System combined, and slightly less than one one-thousandth the mass of the Sun.
                          </span>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="saturn"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-saturn"
                        id="option-1"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-saturn"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Saturn
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="uranus"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-uranus"
                        id="option-2"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-uranus"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Uranus
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="neptune"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-neptune"
                        id="option-3"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-neptune"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Neptune
                            </span>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
                  >
                    <div
                      class="uv_5kpm8k5"
                    >
                      <button
                        class=" uv_5kpm8k3"
                        data-selectgroup="false"
                        data-testid="group-1"
                        role="group"
                        tabindex="-1"
                        type="button"
                      >
                        <span
                          class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owi uv_m4c9ows uv_m4c9ow1b uv_m4c9ow20 uv_m4c9ow3c"
                          style="--uv_qabug40: left;"
                        >
                          TERRESTRIAL PLANETS
                        </span>
                      </button>
                    </div>
                    <div
                      class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u21 uv_toi52u5d"
                      id="items"
                    >
                      <div
                        aria-disabled="false"
                        aria-label="mercury"
                        aria-selected="true"
                        class="uv_5kpm8k6 uv_5kpm8k7"
                        data-testid="option-mercury"
                        id="option-0"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-mercury"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Mercury
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="venus"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-venus"
                        id="option-1"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-venus"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Venus
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="earth"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-earth"
                        id="option-2"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-earth"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Earth
                            </span>
                          </div>
                          <span
                            class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9own uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
                            style="--uv_qabug40: left;"
                          >
                            Our home planet
                          </span>
                        </div>
                      </div>
                      <div
                        aria-disabled="true"
                        aria-label="mars"
                        aria-selected="false"
                        class="uv_5kpm8k6 uv_5kpm8k8"
                        data-testid="option-mars"
                        id="option-3"
                        role="option"
                        tabindex="-1"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-mars"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Mars
                            </span>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-disabled="false"
                        aria-label="pluto"
                        aria-selected="false"
                        class="uv_5kpm8k6"
                        data-testid="option-pluto"
                        id="option-4"
                        role="option"
                        tabindex="0"
                      >
                        <div
                          class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
                          data-testid="option-stack-pluto"
                        >
                          <div
                            class="uv_5kpm8kf uv_toi52u0 uv_toi52u31 uv_toi52u2j uv_toi52u7d uv_toi52u1v uv_toi52u6p"
                          >
                            <span
                              class="uv_5kpm8ke uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow3c"
                              style="--uv_qabug40: left;"
                            >
                              Pluto
                            </span>
                          </div>
                          <span
                            class="uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9own uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
                            style="--uv_qabug40: left;"
                          >
                            Pluto does not fit the usual classification of either terrestrial or Jovian planets, but is rocky
                          </span>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectInputField > should render correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_1_"
          aria-describedby="_r_1_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_1_"
              aria-expanded="false"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8"
              data-disabled="false"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_0_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7knd3 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9owm uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
              >
                Select item
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_34__1sbwqkz1i"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectInputField > should render correctly disabled 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_6_"
          aria-describedby="_r_6_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_6_"
              aria-expanded="false"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8 uv_1ba7knde"
              data-disabled="true"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_5_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7knd3 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9owm uv_m4c9ow1a uv_m4c9ow33"
              >
                Select item
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_true__1sbwqkze styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_2__1sbwqkzm"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectInputField > should render correctly grouped 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_g_"
          aria-describedby="_r_g_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_g_"
              aria-expanded="false"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8"
              data-disabled="false"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_f_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7knd3 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9owm uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
              >
                Select item
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_34__1sbwqkz1i"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectInputField > should render correctly multiselect 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_b_"
          aria-describedby="_r_b_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_b_"
              aria-expanded="false"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8"
              data-disabled="false"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_a_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7knd3 uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owl uv_m4c9owm uv_m4c9ow1b uv_m4c9ow23 uv_m4c9ow3f"
              >
                Select item
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_34__1sbwqkz1i"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;

exports[`selectInputField > should trigger events 1`] = `
<DocumentFragment>
  <div
    data-testid="testing"
  >
    <form
      novalidate=""
      style="display: contents;"
    >
      <div
        aria-label="test"
        class="uv_uf6pag0"
      >
        <div
          aria-controls="_r_29_"
          aria-describedby="_r_29_"
          class="uv_w40vpo9 uv_w40vpob"
          tabindex="-1"
        >
          <div
            class="uv_toi52u0 uv_toi52u31 uv_toi52u2d uv_toi52u7d uv_toi52u1v uv_toi52u5d"
          >
            <div
              aria-controls="_r_29_"
              aria-expanded="false"
              aria-label=""
              class="uv_1ba7knd5 uv_1ba7knd4 uv_1ba7knd9 uv_1ba7knd8"
              data-disabled="false"
              data-readonly="false"
              data-testid="select-input-test"
              id="_r_28_"
              role="combobox"
              tabindex="0"
            >
              <span
                class="uv_1ba7kndl uv_m4c9ow0 uv_m4c9ow2 uv_m4c9ow4 uv_m4c9ow6 uv_m4c9ow8 uv_m4c9owb uv_m4c9owi uv_m4c9owm uv_m4c9ow1b uv_m4c9ow20 uv_m4c9ow3c"
              >
                Mercury
              </span>
              <div
                class="uv_1ba7knd2 uv_toi52u0 uv_toi52u3d uv_toi52u2j uv_toi52u7d uv_toi52u7 uv_toi52u5d"
              >
                <svg
                  aria-label="show dropdown"
                  class="styles__1sbwqkz0 styles_prominence_default__1sbwqkzg styles_disabled_false__1sbwqkzf styles_sentiment_neutral__1sbwqkz3 styles_size_small__1sbwqkza styles_undefined_compound_34__1sbwqkz1i"
                  viewBox="0 0 16 16"
                >
                  <path
                    clip-rule="evenodd"
                    d="M4.22 6.22a.75.75 0 0 1 1.06 0L8 8.94l2.72-2.72a.75.75 0 1 1 1.06 1.06l-3.25 3.25a.75.75 0 0 1-1.06 0L4.22 7.28a.75.75 0 0 1 0-1.06"
                    fill-rule="evenodd"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
